test/generate_tests.py

```diff
@@ -156,3 +156,69 @@ def test_main_drizzle(self):
 			os.remove(test_cosmo_folder+'npy_files/img_%d.npy'%(i))
 		os.rmdir(test_cosmo_folder+'npy_files')
 		os.rmdir(output_folder)
+
+	def test_h5(self):
+		# Test that the h5 kwarg generates the required h5 files
+		old_sys = copy.deepcopy(sys.argv)
+		n_generate = 21
+		output_folder = 'test_data/test_dataset'
+		sys.argv = ['test','test_data/config_dict.py',output_folder,
+			'--n',str(n_generate),'--h5']
+		if tensorflow_installed:
+			sys.argv.append('--tf_record')
+		generate.main()
+
+		image_file_list = glob.glob(os.path.join(output_folder,'image_data.h5'))
+		# Check only 1 h5 file is made:
+		self.assertEqual(len(image_file_list),1)
+		with h5py.File(image_file_list[0],'r') as f0:
+			image_data = f0['data'][()]
+		image_shape = image_data.shape
+		print('image shape',image_data.shape)
+		self.assertEqual(image_shape[0],n_generate)
+
+		# Make sure all of the files are readable and have the correct size
+		# for the config
+		for n in range(n_generate):
+			img = image_data[n,:,:]
+			self.assertTupleEqual(img.shape,(64,64))
+		os.remove(image_file_list[0])
+
+		# Make sure the metadata makes sense
+		metadata = pd.read_csv(os.path.join(output_folder,'metadata.csv'))
+		self.assertEqual(len(metadata),n_generate)
+		self.assertListEqual(list(
+			metadata['cosmology_parameters_cosmology_name']),
+			['planck18']*n_generate)
+		self.assertListEqual(list(
+			metadata['detector_parameters_pixel_scale']),[0.08]*n_generate)
+		self.assertListEqual(list(metadata['subhalo_parameters_c_0']),
+			[18.0]*n_generate)
+		self.assertListEqual(list(metadata['main_deflector_parameters_z_lens']),
+			[0.5]*n_generate)
+		self.assertListEqual(list(metadata['source_parameters_z_source']),
+			[1.5]*n_generate)
+		self.assertGreater(np.std(metadata['source_parameters_catalog_i']),0)
+		self.assertListEqual(list(metadata['psf_parameters_fwhm']),
+			[0.1]*n_generate)
+		# Check that the subhalo_parameters_sigma_sub are being drawn
+		self.assertGreater(np.std(metadata['los_parameters_delta_los']),
+			0.0)
+		# Check that nothing is getting written under cross_object
+		for key in metadata.keys():
+			self.assertFalse('cross_object' in key)
+			self.assertFalse('source_exclusion_list' in key)
+
+		# Remove the metadata file
+		os.remove(os.path.join(output_folder,'metadata.csv'))
+		os.remove(os.path.join(output_folder,'config_dict.py'))
+
+		sys.argv = old_sys
+
+		# Also clean up the test cosmos cache
+		test_cosmo_folder = 'test_data/cosmos/'
+		
```

Comment:
You'll also want to remove the h5 file here; currently the test fails when it tries to remove the non-empty directory in the final rmdir call.